angela-angetam/hw_2/5-random-address-generator-bonus/app.js

```diff
@@ -0,0 +1,32 @@
+/*
+Optional bonus work on conditionals and functions:
+
+    Write a script that can generate random addresses
+    As a first step, create arrays that contain dummy data for each of the following: street number, street name, city name, state name, and zip code
+    Your script should randomly select one item from each of these arrays and then use them to construct a random address
+    Each time you load or reload the page in a browser, your program should print a new randomly-generated address to the terminal. For example:
+    
+    34578 Dolphin Street, Wonka NY, 44506
+*/
+var i;
+var stNum = [12345, 52345, 12575, 12453];
+var stName = ['Apple Street', 'Pear Way', 'Pineapple Lane', 'Grape Street'];
+var city = ['San Francisco', 'San Mateo', 'Oakland', 'Berkeley'];
+var state = ['CA', 'NV', 'NY', 'OR'];
+var zip = [55555, 66666, 77777, 88888];
+var i = Math.floor(Math.random() * stNum.length);
+
+function randAddress() {
+    console.log(stNum[i] + " " + stName[i] + ", " + city[i] + ", " + state[i] + " " + zip[i]);
+};
+randAddress(); 
+/*
+Is it possible to do an array of arrays and then use the foreach so a random number is generated for each array?
```

Comment:
Absolutely! You can loop inside of a loop, and you can loop inside of that loop, and loop what's in that loop. The general best practice, however, is that if you need to nest a loop you should only nest it one loop deep; with each loop your code becomes exponentially slower.

Give it a shot. ⭐️